sharding/indexHashedNodesCoordinator.go

```diff
@@ -604,6 +594,35 @@ func (ihgs *indexHashedNodesCoordinator) EpochStartPrepare(metaHdr data.HeaderHa
 	ihgs.consensusGroupCacher.Clear()
 }
 
+func (ihgs *indexHashedNodesCoordinator) fillPublicKeyToValidatorMap() {
+	ihgs.mutNodesConfig.Lock()
+	defer ihgs.mutNodesConfig.Unlock()
+
+	index := 0
+	epochList := make([]uint32, len(ihgs.nodesConfig))
+	mapAllValidators := make(map[uint32]map[string]*validatorWithShardID)
+	for epoch, epochConfig := range ihgs.nodesConfig {
+		epochConfig.mutNodesMaps.RLock()
+		mapAllValidators[epoch] = ihgs.createPublicKeyToValidatorMap(epochConfig.eligibleMap, epochConfig.waitingMap)
+		epochConfig.mutNodesMaps.RUnlock()
+
+		epochList[index] = epoch
+		index++
+	}
+
+	sort.Slice(epochList, func(i, j int) bool {
+		return epochList[i] < epochList[j]
```

Comment:
:+1: that should overwrite the values from previous epochs, if the node is performing well.

Reply:
That was the idea